crates/rpc/rpc/src/debug.rs

```diff
@@ -479,6 +480,8 @@ where
                         }
                         results.push(trace);
                     }
+                    block_env.number += U256::from(1);
```

Comment:
it's safer to get the block `number` and `timestamp` from `block`. otherwise the debug help here is made up and may be out of sync with what is actually being debugged - this can result in super hard bugs to squash since by default we will trust log output/docs/code comments!

Reply:
or maybe it's enough to check the block env number and timestamp matches with the block before updating like this. possibly I'm missing if this check is nested in some call in this scope?

Reply:
this adds the intended behaviour re timestamp+number across multiple bundles:

https://www.chainnodes.org/docs/ethereum/debug_traceCallMany